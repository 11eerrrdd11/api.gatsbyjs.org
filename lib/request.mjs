import axios from 'axios';

export const put = (uri, payload, config = {}) =>
  axios.put(uri, payload, config);

export const post = (uri, payload, config = {}) =>
  axios.post(uri, payload, config);
