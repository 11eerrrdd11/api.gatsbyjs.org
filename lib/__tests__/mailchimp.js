import * as request from '../request.mjs';
import { addMailChimpSubscriber } from '../mailchimp.mjs';

jest.mock('dotenv', () => ({
  config: () => {
    // Provide the env vars expected by the lib
    process.env.MAILCHIMP_API_URI = 'https://us001.api.mailchimp.com/3.0';
    process.env.MAILCHIMP_LIST_ID = 'TEST_MAILCHIMP_LIST_ID';
    process.env.MAILCHIMP_API_KEY = 'TEST_MAILCHIMP_API_KEY';
  }
}));

let mockReturn = () => {};
const mockPayload = {
  username: 'testuser',
  email: 'test@example.org',
  first_name: 'Test',
  last_name: 'User'
};

jest.mock('../request.mjs', () => ({
  put: jest.fn(() => mockReturn)
}));

describe('lib/mailchimp', () => {
  test('calls the correct endpoint with the correct payload', () => {
    addMailChimpSubscriber(mockPayload);
    expect(request.put.mock).toMatchSnapshot();
  });
});
