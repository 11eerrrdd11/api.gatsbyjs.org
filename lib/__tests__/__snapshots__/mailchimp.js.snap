// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/mailchimp calls the correct endpoint with the correct payload 1`] = `
Object {
  "calls": Array [
    Array [
      "https://us001.api.mailchimp.com/3.0/lists/TEST_MAILCHIMP_LIST_ID/members/0c17bf66e649070167701d2d3cd71711",
      Object {
        "email_address": "test@example.org",
        "email_type": "html",
        "merge_fields": Object {
          "FNAME": "Test",
          "GITHUB": "testuser",
          "LNAME": "User",
        },
        "status_if_new": "subscribed",
      },
      Object {
        "auth": Object {
          "password": "TEST_MAILCHIMP_API_KEY",
          "username": "apikey",
        },
      },
    ],
  ],
  "instances": Array [
    undefined,
  ],
  "invocationCallOrder": Array [
    1,
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": [Function],
    },
  ],
}
`;
